lib/matplotlib/tests/test_path.py

```diff
@@ -45,6 +45,23 @@ def test_contains_points_negative_radius():
     assert np.all(result == expected)
 
 
+def test_point_in_path_nan():
+    box = np.array([[0, 0], [1, 0], [1, 1], [0, 1], [0, 0]])
+    p = Path(box)
+    test = np.array([[np.nan, 0.5]])
+    contains = p.contains_points(test)
+    assert len(contains) == 1
+    assert not contains[0]
+
+
+def test_nonlinear_containment():
+    fig, ax = plt.subplots()
+    ax.set(xscale="log", ylim=(0, 1))
+    polygon = ax.axvspan(1, 10)
+    assert polygon.get_path().contains_point(
+        ax.transData.transform_point((5, .5)), ax.transData)
+
```

Comment:
May also be worth checking that the points (0.5, 0.5) and (20, 0.5) are not inside `polygon`